specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2020-12-01/Marketplace.json

```diff
@@ -0,0 +1,1068 @@
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AdminRequestApprovalsResource"
+            }
+          },
+          "default": {
+            "description": "Microsoft.Marketplace error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "UpdateAdminRequestApproval": {
+            "$ref": "./examples/UpdateAdminRequestApproval.json"
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "PrivateStoreIdParameter": {
+      "name": "privateStoreId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "description": "The store ID - must use the tenant ID"
+    },
+    "ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "The API version to use for the request."
+    },
+    "PublisherIdParameter": {
+      "name": "publisherId",
+      "in": "query",
+      "required": true,
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "description": "The publisher id of this offer."
+    },
+    "OfferIdPathParameter": {
+      "in": "path",
+      "name": "offerId",
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "required": true,
+      "description": "The offer ID to update or delete"
+    },
+    "AdminRequestApprovalIdPathParameter": {
+      "in": "path",
+      "name": "adminRequestApprovalId",
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "required": true,
+      "description": "The admin request approval ID to get create or update"
+    },
+    "RequestApprovalIdPathParameter": {
+      "in": "path",
+      "name": "requestApprovalId",
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "required": true,
+      "description": "The request approval ID to get create or update"
+    }
+  },
+  "definitions": {
+    "ErrorResponse": {
+      "description": "Error response indicates Microsoft.Marketplace service is not able to process the incoming request. The reason is provided in the error message.",
+      "type": "object",
+      "properties": {
+        "error": {
+          "description": "The details of the error.",
+          "properties": {
+            "code": {
+              "description": "Error code.",
+              "type": "string",
+              "readOnly": true
+            },
+            "message": {
+              "description": "Error message indicating why the operation failed.",
+              "type": "string",
+              "readOnly": true
+            }
+          }
+        }
+      }
+    },
+    "PrivateStoreNotificationsState": {
```

Comment:
For all these array properties, do you have a sense on how many in average the service will return? Reason I ask is because if the list is pretty long you might want to move these properties as independent GET calls with pagination. If the list can grow large, can you please ack this request and create a work item to track this work in a future api version?

Reply:
created WI: Task 9250334: Consider "independent GET calls with pagination" - Boards (visualstudio.com)

Reply:
[https://msazure.visualstudio.com/Marketplace/_workitems/edit/9250334](url)